scripts/validate_docstrings.py

```diff
@@ -471,6 +471,45 @@ def parameter_desc(self, param):
                 desc = desc[:desc.index(full_directive)]
         return desc
 
+    def parameter_desc_list(self, param):
+        """Return a parameter description as a list.
+
+        Each line of the parameter docstring is a string in the list.
+        """
+        i = list(self.doc_parameters).index(param)
+        desc_list = [line
+                     for line in self.doc._parsed_data['Parameters'][i][-1]
+                     if line]
+        # Find and strip out any sphinx directives
+        for directive in DIRECTIVES:
+            full_directive = '.. {}'.format(directive)
+            for desc_item in desc_list:
+                if full_directive in desc_item:
+                    # Only retain any description before the directive
+                    desc_list = desc_list[:desc_list.index(desc_item)]
+                    break
+        return desc_list
+
+    def parameter_correct_end(self, param):
+        """Return True if a parameter description is terminated correctly.
+
+        A parameter description should always be terminated with a period, a
+        part from when it ends with a bullet point.
+        """
+        if self.parameter_desc(param)[-1] == '.':
+            return True
+        else:
+            # Return True if a parameter description ends
+            # with a bullet point
+            for desc_line in self.parameter_desc_list(param)[::-1]:
+                if desc_line[0] in ['*', '-', '+', '•', '‣', '⁃']:
+                    return True
+                elif desc_line[0:2] != '  ':
```

Comment:
What is this condition for?

Reply:
The method enters in this for-loop only when the description does not end with a period.

It then checks the description line by line backwards, if the first character of the line is a bullet point returns `True` straightaway, if the first 2 character of the line are not 2 blank spaces instead returns `False` since the last line is not part of a bullet point for sure. If instead the first 2 character are 2 blank spaces, it needs to examine the 2nd to last line to check that the last line is part of a bullet point and so on.

When the description of the parameter is longer that 1 line, the last line (`desc_line`) starts with 2 blank spaces (for indentations).
Example:
```
* 'max': the description for max that can also be very long and
   requires an extra row
```
becomes after calling `self.parameter_desc_list(param)`
```
["* 'max': the description for max that can also be very long and",
 "  requires an extra row", ...]
```